### ISSUE TITLE
Caught an AmazonServiceException, which means your request made it to Amazon S3, but was rejected with an error response for some reason.

### ISSUE THREAD

**Issue by NajibAbuSamra** [shalomweiss/mm-system-2017#84]
> Caught an AmazonServiceException, which means your request made it to Amazon S3, but was rejected with an error response for some reason.
> Error Message:    The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: 892A9C09B5398F2F; S3 Extended Request ID: trjwQ+GK9BjK7rGIG4NiO5SsWw8okOmiYMK6FGN3wyWr///4QZH5vAICAH+wCdRivrHqg0dm+UQ=)
> HTTP Status Code: 404
> AWS Error Code:   NoSuchKey
> Error Type:       Client
> Request ID:       892A9C09B5398F2F 
> 

 
This happens on the Mentors page and Mentee page

**Comment by masterFoad** [comment]
@NajibAbuSamra can you post the request that you sent?
or the action that led to this error?

**Comment by masterFoad** [comment]
@NajibAbuSamra oh just noticed "This happens on the Mentors page and Mentee page".

**Comment by NajibAbuSamra** [comment]
I don't think this happens from the GetAllMentors servlet, I think it is caused by trying to access the S3 data (cv/grades/profilepic/etc.) but I have no idea why it says the key is wrong, I checked and it's the last key you sent and I don't think it should have changed since.
